src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/HandlerMethodDescriptor.cs

```diff
@@ -1,12 +1,16 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Reflection;
+using System.Threading.Tasks;
 
 namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
 {
     public class HandlerMethodDescriptor
     {
         public MethodInfo Method { get; set; }
+
+        public Func<Page, object, Task<IActionResult>> Executor { get; set; }
```

Comment:
This is a change in the right direction so I think we should get it in, but we might want this caching to be part of the invoker.